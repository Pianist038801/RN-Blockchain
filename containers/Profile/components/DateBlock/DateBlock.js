import React from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'
import styles from '../../../../style.js'

const DateBlock = (props) => {
  let { date } = props
  date = new Date(date)

  return (
    <View style={styles.profileEntityRight}>
      <Text style={styles.profileEntityDate}>{date.getFullYear()}</Text>
    </View>
  )
}

DateBlock.propTypes = {
  date: PropTypes.string.isRequired
}

export default DateBlock
