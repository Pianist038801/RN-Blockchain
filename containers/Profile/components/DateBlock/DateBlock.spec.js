import React from 'react'
import renderer from 'react-test-renderer'
import DateBlock from './index'

describe('containers/Profile/components/DateBlock', () => {
  it('renders correctly', () => {
    const props = {
      date: '2017-11-04'
    }

    const tree = renderer.create(
      <DateBlock {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
