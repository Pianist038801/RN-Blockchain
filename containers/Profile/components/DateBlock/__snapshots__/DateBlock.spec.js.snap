// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/DateBlock renders correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    2017
  </Text>
</View>
`;
